### PR TITLE
Group with_preset methods into ModulePreset trait

### PR DESCRIPTION
**Pull request by jakelang**

### CODE REVIEW

File: libchisel/src/lib.rs [hunk 1]
```diff
@@ -37,6 +37,12 @@ pub trait ModuleValidator {
     fn validate(&self, module: &Module) -> Result<bool, ModuleError>;
 }
 
+pub trait ModulePreset {
+    fn with_preset(&str) -> Result<Self, ()>
```
Comment (axic) [inline libchisel/src/lib.rs]: I think we could have an `enum` for settings, which is then extended by the modules, just like how `ModuleError` works.

File: libchisel/src/trimexports.rs [hunk 2]
```diff
@@ -40,6 +33,15 @@ impl ExportWhitelist {
             _ => Err(()),
         }
     }
+}
+
+impl ExportWhitelist {
```
Comment (axic) [inline libchisel/src/trimexports.rs]: Why move this?

File: libchisel/src/verifyimports.rs [hunk 3]
```diff
@@ -281,7 +281,9 @@ impl<'a> VerifyImports<'a> {
             _ => Err(()),
         }
     }
+}
 
+impl<'a> VerifyImports<'a> {
```
Comment (axic) [inline libchisel/src/verifyimports.rs]: Could wrap this entire implementation into `[cfg(test)]`, though if you do that, please do in a separate commit.

File: libchisel/src/trimexports.rs (hunk 2 above)
Comment (jakelang) [reply libchisel/src/trimexports.rs]: Out of habit I put trait impls before method impls.

File: libchisel/src/lib.rs (hunk 1 above)
Comment (jakelang) [reply libchisel/src/lib.rs]: Can you give an example of such a setting enum? I think that instantiating modules with the preset as a string is more robust when considering the possibility of third party modules as well as nonstandard presets.

File: libchisel/src/verifyimports.rs (hunk 3 above)
Comment (jakelang) [reply libchisel/src/verifyimports.rs]: will do

File: libchisel/src/lib.rs (hunk 1 above)
Comment (axic) [reply libchisel/src/lib.rs]: `enum` for the optional settings, such as `payload` in `deployer`.

File: libchisel/src/verifyimports.rs (hunk 3 above)
Comment (jakelang) [reply libchisel/src/verifyimports.rs]: Done

File: libchisel/src/lib.rs (hunk 1 above)
Comment (jakelang) [reply libchisel/src/lib.rs]: hmm. I'm starting to think that once configuration becomes a bit complex we should have a dedicated context structure
